src/txn/txn_ckpt.c

```diff
@@ -627,6 +629,22 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
 
     __wt_writeunlock(session, &txn_global->rwlock);
 
+    /*
+     * Open an internal session to allocate a transaction id that is used for removing the history
```

Comment:
"Open an internal session to allocate a transaction id that will be used for removing history entries. This is used when a prepare transaction rollback occurs in parallel to a checkpoint. Ensure that this transaction id is published before taking the checkpoint's snapshot."

Reply:
Fixed.